sdk/template/Azure.Template/src/Generated/Models/DerivedFromBaseClassWithDiscriminatorA.Serialization.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Text.Json;
+
+namespace Azure.Template.Models
+{
+    public partial class DerivedFromBaseClassWithDiscriminatorA
+    {
+        internal override void WriteCore(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("discriminatorProperty");
+            writer.WriteStringValue(DiscriminatorProperty);
```

Comment:
I still don't understand why do we need `DiscriminatorProperty`. What prevents us from hardcoding it:
```cs
writer.WritePropertyName("discriminatorProperty");
writer.WriteStringValue("A");
```